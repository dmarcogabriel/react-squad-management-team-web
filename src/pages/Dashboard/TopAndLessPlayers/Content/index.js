import React from 'react';
import cn from 'classnames';
import { showPlayerInitials } from '../../../../utils/playersInitials';
import PlayerAvatarPlaceholder from '../../../../assets/playerAvatarPlaceholder.png';

const Content = ({ playerName, title, percentage, className, dataTestid }) => (
  <div
    data-testid={dataTestid || 'className'}
    className={cn('w-full p-5', className)}
  >
    <h2 data-testid="title" className="font-bold text-lg text-center">
      {title}
    </h2>

    <div className="relative flex justify-center my-3">
      {playerName ? (
        <div
          className={cn(
            'relative p-12 bg-white',
            'text-primary-dark rounded-full'
          )}
        >
          <p
            data-testid="playerName"
            className={cn(
              'text-3xl absolute w-full h-full',
              'flex justify-center items-center',
              'top-0 left-0 right-0'
            )}
          >
            {showPlayerInitials(playerName)}
          </p>
        </div>
      ) : (
        <div
          className={cn(
            'relative p-12 border-dashed border-2',
            'text-primary-dark rounded-full overflow-hidden'
          )}
        >
          <img
            className="absolute right-0 top-0 w-full h-full"
            src={PlayerAvatarPlaceholder}
            alt="Player Avatar"
          />
        </div>
      )}

      <p
        data-testid="percentage"
        className={cn(
          'absolute right-0 top-0',
          'text-base border-b-2 border-white pr-5'
        )}
      >
        {percentage}
      </p>
    </div>
  </div>
);

export default Content;
