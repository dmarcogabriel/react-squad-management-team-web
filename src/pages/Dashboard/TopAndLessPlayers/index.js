import React, { useState, useEffect } from 'react';
import lodash from 'lodash';
import { usePlayers } from '../../../contexts/Players';
import Content from './Content';

const TopAndLessPlayers = () => {
  const [mostPickedPlayer, setMostPickedPlayer] = useState(null);
  const [lessPickedPlayer, setLessPickedPlayer] = useState(null);

  const [players] = usePlayers();

  useEffect(() => {
    if (players && players.length) {
      const allPlayers = lodash.groupBy(players, 'player_id');

      const a = Object.keys(allPlayers).map((k) => allPlayers[k]);

      const [mostPicked] = a.reduce((b, c) => (b.length > c.length ? b : c));

      const [lessPicked] = a.reduce((b, c) => (b.length > c.length ? c : b));

      setMostPickedPlayer(mostPicked);
      setLessPickedPlayer(lessPicked);
    }
  }, []);

  return (
    <div className="rounded-lg flex flex-col md:flex-row shadow-lg bg-gradient-to-t from-secondary-dark to-secondary-light text-white mb-5">
      <Content
        className="border-b-2 md:border-r-2 md:border-b-0 border-opacity-25 border-white"
        playerName={mostPickedPlayer && mostPickedPlayer.display_name}
        percentage="0%"
        title="Most picked player"
      />

      <Content
        playerName={lessPickedPlayer && lessPickedPlayer.display_name}
        percentage="0%"
        title="Less picked player"
      />
    </div>
  );
};

export default TopAndLessPlayers;
