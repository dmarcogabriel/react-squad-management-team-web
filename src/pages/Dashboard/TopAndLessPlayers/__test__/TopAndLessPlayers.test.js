import { cleanup } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import TopAndLessPlayers from '..';

describe('TopAndLessPlayers tests', () => {
  afterEach(cleanup);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TopAndLessPlayers />, div);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<TopAndLessPlayers />);

    expect(wrapper).toMatchSnapshot();
  });
});
