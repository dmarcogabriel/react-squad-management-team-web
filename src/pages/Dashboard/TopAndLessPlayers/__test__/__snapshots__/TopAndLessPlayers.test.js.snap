// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopAndLessPlayers tests matches snapshot 1`] = `
<div
  className="rounded-lg flex flex-col md:flex-row shadow-lg bg-gradient-to-t from-secondary-dark to-secondary-light text-white mb-5"
>
  <Content
    className="border-b-2 md:border-r-2 md:border-b-0 border-opacity-25 border-white"
    percentage="0%"
    playerName={null}
    title="Most picked player"
  />
  <Content
    percentage="0%"
    playerName={null}
    title="Less picked player"
  />
</div>
`;
